jlm/llvm/ir/operators/MemoryStateOperations.cpp

```diff
@@ -279,6 +282,30 @@ LambdaExitMemoryStateMergeOperation::copy() const
   return std::make_unique<LambdaExitMemoryStateMergeOperation>(*this);
 }
 
+rvsdg::Input *
+LambdaExitMemoryStateMergeOperation::MapMemoryNodeIdToInput(
+    const rvsdg::Node & node,
+    const MemoryNodeId memoryNodeId)
+{
+  const auto operation =
+      dynamic_cast<const LambdaExitMemoryStateMergeOperation *>(&node.GetOperation());
+  if (!operation)
+  {
+    return nullptr;
+  }
+
+  try
+  {
+    const auto index = operation->MemoryNodeIdToIndex_.LookupKey(memoryNodeId);
```

Comment:
I guess doing separate calls to `HasKey` and `LookupKey` would be slower. I am not really used to using `try`/`catch`, but I guess it makes sense here

Reply:
I can change it. I just wanted to avoid the duplicated access to the map.

Reply:
Changed.